lib/mongoid/extensions/hash.rb

```diff
@@ -217,10 +217,12 @@ module ClassMethods
         #
         # @param [ Object ] object The object to mongoize.
         #
-        # @return [ Hash ] The object mongoized.
+        # @return [ Hash | nil ] The object mongoized or nil.
         def mongoize(object)
           return if object.nil?
-          evolve(object.dup).transform_values!(&:mongoize)
+          if object.is_a?(Hash)
+            evolve(object.dup).transform_values!(&:mongoize)
```

Comment:
Don't we also need to mongoize the Hash keys?

Reply:
Keys in BSON documents must be strings, see https://bsonspec.org/spec.html. Non-string keys are not going to round-trip.

Reply:
Perhaps `#mongoize` should call `.deep_stringify_keys!` here then?